### PR TITLE
회원가입 이메일 인증과 메일 아웃박스 구조 정리

### PR DESCRIPTION
**Pull request by minyongP**
# 한줄 요약
- 회원가입 흐름을 `이메일 중복 확인 -> 인증 요청 -> 인증 완료` 3단계로 변경했습니다.
- 메일 발송을 `mail_outbox` 기반 비동기 처리로 전환했습니다.
- 메일 아웃박스 구조를 `Notifier -> Outbox -> Scheduler -> Service -> Handler -> Template -> Sender` 흐름으로 정리했습니다.
- `SmtpMailSender` warn 로그와 스케줄러 thread pool 설정을 추가했습니다.

# 관련 작업 / 이슈
- #56
- #83

---

### 1. 회원가입 인증 흐름 변경
- 기존 즉시 계정 생성 방식 대신 `pending_signup`을 만들고 인증 완료 시점에만 실제 계정을 생성하도록 변경했습니다.
- 동일 이메일 재요청 시 새 가입 세션을 만드는 대신 기존 `pending_signup`을 재사용하고 인증 코드와 만료 시간만 갱신하도록 바꿨습니다.
- 만료된 회원가입 대기 건을 정리하는 cleanup batch를 추가했습니다.

### 2. 메일 발송을 outbox 기반 비동기 처리로 전환
- 회원가입 인증 메일과 워크스페이스 초대 메일 모두 `mail_outbox` 적재 후 배치 워커가 전송합니다.
- API 성공 시점과 실제 SMTP 전송 시점을 분리해 메일 실패가 비즈니스 트랜잭션을 깨지 않도록 변경했습니다.
- `mail_outbox` 테이블과 상태/재시도 로직을 추가했습니다.

### 3. 메일 아웃박스 구조 정리
- `MailNotifier`가 메일 요청 포트 역할을 담당합니다.
- `OutboxMailNotifier`가 outbox 적재 구현을 담당합니다.
- `MailOutboxScheduler`가 due outbox를 주기적으로 선점/처리합니다.
- `MailOutboxService`가 reserve / recover / deliver / retry 흐름을 제어합니다.
- `MailOutboxHandler`는 메일 타입별 payload를 발송 가능한 메시지로 변환합니다.
- 템플릿은 타입별 renderer로 분리했습니다.
  - `SignupVerificationMailTemplateRenderer`
  - `WorkspaceInviteMailTemplateRenderer`
  - 공통 레이아웃은 `MailTemplateLayout`
- 실제 SMTP 연동은 `MailSender` 포트 뒤의 `SmtpMailSender`가 담당합니다.

### 4. 운영 / 추적 보강
- `SmtpMailSender`에서 SMTP 인증 실패 / 전송 실패 / 일반 메일 예외를 조금 더 구분해 `warn` 로그를 남기도록 보강했습니다.
- scheduler 전용 `ThreadPoolTaskScheduler`를 추가했습니다.
- `app.scheduler.pool-size` (`APP_SCHEDULER_POOL_SIZE`, 기본값 `4`) 설정으로 스케줄 작업이 서로 덜 막히도록 조정했습니다.

### 5. DB / 마이그레이션
- `pending_signups`, `mail_outbox` 테이블 생성 마이그레이션을 추가했습니다.

---

## 커밋 구성
- `refactor : 메일 아웃박스 발송 구조 정리`
- `chore : 스케줄러 스레드 풀 설정 추가`
- `docs : 메일 아웃박스 클래스 플로우 SVG 추가`

---

## 테스트
- `./gradlew test --tests com.ujax.application.mail.template.SignupVerificationMailTemplateRendererTest --tests com.ujax.application.mail.template.WorkspaceInviteMailTemplateRendererTest --tests com.ujax.application.mail.outbox.MailOutboxServiceTest --tests com.ujax.infrastructure.external.mail.SmtpMailSenderTest`
- `./gradlew test --tests com.ujax.infrastructure.config.SchedulingConfigTest --tests com.ujax.infrastructure.scheduling.mail.MailOutboxSchedulerTest --tests com.ujax.infrastructure.scheduling.auth.PendingSignupCleanupSchedulerTest`

---

## PR 체크리스트
- [x] 관련 이슈(작업 항목)를 PR 설명에 연결했다.
- [x] 주요 회귀 테스트를 로컬에서 실행해 모두 통과했다.
- [x] 불필요한 로그, 디버깅 코드, 주석을 제거했다.
- [x] 이 변경과 충돌할 수 있는 다른 PR이나 변경 사항이 없는지 확인했다.

---

## Breaking Change 여부
- [x] Yes (호환성 깨짐 있음)
- [ ] No

### 배포 / 마이그레이션 시 주의사항
- 기존 단일 회원가입 API 호출 방식은 더 이상 그대로 사용할 수 없습니다.
- 클라이언트는 `이메일 중복 확인 -> 인증 요청 -> 인증 완료` 순서로 연동을 변경해야 합니다.
- `pending_signups`, `mail_outbox` 테이블 생성 마이그레이션이 선행되어야 합니다.
- 메일 발송은 비동기 처리로 바뀌어 API 성공 시점과 실제 메일 도착 시점이 분리됩니다.
- 스케줄러 pool size는 `APP_SCHEDULER_POOL_SIZE`로 조정할 수 있으며 기본값은 `4`입니다.

---

## 기타 정보 / 후속 고려사항
- outbox 워커를 멀티 인스턴스로 확장할 경우 row claim 동시성 보강이 필요합니다.
- signup verification resend 제한 및 인증 시도 횟수 제한(throttling) 정책은 후속 작업으로 남아 있습니다.
- webhook timeout / webhook batch 설정 검토는 별도 브랜치에서 진행 예정입니다.

---

## 클래스 플로우
메일 아웃박스 클래스 플로우

### CODE REVIEW

**Review by kwongwangjae — COMMENTED** [review]
- 내용이 많아서 내일 밤 회의전까지 다 읽어보고 올께!
- 흐름 같이 보면 좋을듯

**Comment by minyongP** [comment]
# deprecated
pr-56-signup-email-outbox-flow

**Comment by minyongP** [comment]
pr-56-client-api-before-after

**Comment by kwongwangjae** [comment]
위에 정리한거 보니까 더 이해하기 쉽네 땡스
완전히 이해했다고 보긴 어렵긴 한데 쫌 내부 로직이 있어서 이해한 수준에서는 좋은듯?? 더 간결해지고, 메일 관리도 잘되고 깔끔하고 

근데 대신 이게 external 패키지에 있어야 하는지 아니면 domain인지 service인지만 쫌 구분해서 잘 나눠주면 좋을듯??
ObjectMapper아니면 선언문들이나 너무 서비스 패키지에 몰려 있고, 쫌 애매한 위치에 있는 것들이 있는듯 그래서 뭔가 복잡해 보여

구현 고생했습니다~~

**Review by kwongwangjae — COMMENTED** [review]
다 되면 편하게 푸쉬해주세여~~
svg 파일은 필요하면 살려두고  없으면 삭제해도 되고 편한대로 하면 될듯??

**Comment by minyongP** [comment]
음 더 진행할건 없는것 같고, hook도 좀 수정할거는 다른 pr로 진행할게요. 프론트 작업 먼저 할까요

**Comment by kwongwangjae** [comment]
응 push해주고, 프론트 기본 작업도 해봐주라 api-spec읽어보고 등록해주면 될듯?? 그냥 restDoc 등록이라 갠간 pr올려주면 잘 되어 있는 지 확인해볼께 프론트
